File: Leetcode/88. Merge Sorted Array.ts
function merge(nums1: number[], m: number, nums2: number[], n: number): void {
    let arr: number[] = [];
    let i=0, j=0;
    while(i<m || j<n) {
        if(i>=m || nums1[i] > nums2[j]) {
            arr.push(nums2[j]);
            j++;
        } else {
            arr.push(nums1[i]);
            i++;
        }
    }

    for(let i=0; i<arr.length; i++) {
        nums1[i] = arr[i];
    }
};

File: Leetcode/Rust/88. Merge Sorted Array.rs
impl Solution {
    pub fn merge(nums1: &mut Vec<i32>, m: i32, nums2: &mut Vec<i32>, n: i32) {
        let (mut i, mut j, mut k) = (m-1, n-1, m+n-1);
        
        while(k >= 0) {
            if(j < 0) { break; }
            else if(i < 0 || nums1[i as usize] < nums2[j as usize]) {
                nums1[k as usize] = nums2[j as usize];
                j -= 1;
            } else {
                nums1[k as usize] = nums1[i as usize];
                i -= 1;
            }
            k -= 1;
        }
    }
}
